msprime/simulations.py

```diff
@@ -2477,3 +2477,20 @@ def num_epochs(self):
         Returns the number of epochs defined by the demographic model.
         """
         return len(self.epochs)
+
+
+def rerun(ts):
+    """
+    Reruns the simulations encoded in the provenance of the specified tree
+    sequence.
+    """
+    # TODO We can imagine passing in extra software/cmd arguments which map to a
+    # callable.
+    # TODO figure out how to workaround the circular import problems. Possibly
+    # this should just be in it's own module or something.
+    cmd_map = {"simulate": simulate}  # , "mutate": mutations.mutate}
```

Comment:
I would suggest that in provenances `command` becomes fully qualified i.e msprime.simulate. Then it can be dynamically imported as long as the pip package is in the users env.